presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -75,3 +81,28 @@ def __validate_position_over_text(self, analyzer_result, text_len):
                 f"Invalid analyzer result: '{analyzer_result}', "
                 f"original text length is only {text_len}."
             )
+
+    def __extract_anonymizer(self, transformation):
+        anonymizer = transformation.get("anonymizer")()
+        # if the anonymizer is not valid, a InvalidParamException
+        anonymizer.validate(params=transformation)
+        return anonymizer
+
+    def __anonymize_section(
+        self, anonymizer, output_text, analyzer_result, transformation
+    ):
+        text_to_anonymize = output_text[analyzer_result.start : analyzer_result.end]
+        anonymized_text = anonymizer.anonymize(
+            params=transformation, text=text_to_anonymize
+        )  # TODO: [ADO-2754] replace with the singleton class instance
+        return anonymized_text
+
+    def __update_output_with_anonymized_section(
+        self, analyzer_result, last_replacement_point, output_text, anonymized_text
```

Comment:
I thought about doing this, but think the method is having too many parameters. 
Perhaps moving it to a class which handles the parameters will be more correct but I think it's not such a small refactor.

Reply:
my intention was to leave the main loop more readable, and keep the helper methods in the class to make it easier to follow up when reading the code

Reply:
I know, but I think passing over 3 parameters to a method makes it messy as well. 
That's why I didn't do the refactor in the first place... I think the best way for this is to create a class that holds the data and knows how to handle it. 
But it doesn't need to be in this review.
We can enter this and do it later.